pymc_marketing/mmm/utility.py

```diff
@@ -0,0 +1,546 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""
+Utility functions for Bayesian optimization.
```

Comment:
shall we call this file `utils.py` for consistency across the package?

Reply:
Actually there are many functions here which we can move to their own module `risk.py`. This way we keep the modules more consistent instead of putting many thing here. wdyt?

Reply:
Yes, talking with @cluhmann and @wd60622 we think that even though most functions are now directed at risk, the truth is that the change is bigger than that. We are allowing utility functions to be changed for any purpose, one could be risk, but the user could define anything. Therefore, calling it **risk** or something is really like closing the scope when it could be much more.

I think _utils_ would not be a suitable name, since we are talking about **utility functions**, which is a bayesian name, basically a similar term for the objective function in the bayesian optimization context. Those functions are not simple `utils` for everything.

What do you think?

Reply:
I am for `utility.py`. I read that different than `utils.py`

Reply:
Ok! Ni strong opinion here. In a different PR we just need to harmonize the term across the package if needed :)